File: backend/Rashtan.AuditStory.Workflows/ServiceCollectionExtension.fs
﻿namespace Rashtan.AuditStory.Workflows

open System.Runtime.CompilerServices
open Microsoft.Extensions.DependencyInjection
open Rashtan.AuditStory.Common

[<Extension>]
type ServiceCollectionExtension =
    [<Extension>]
    static member RegisterWorkflows(services: IServiceCollection) =
        services.AddSingleton<IDateTimeProvider, DateTimeProvider>()
                .AddSingleton<IUserStatusWorkflow, UserStatusWorkflow>()
                .AddSingleton<UserProfileWorkflow>()                
                .AddSingleton<CompanyWorkflow>()
                .AddSingleton<CategoryWorkflow>()                
                .AddSingleton<CountryWorkflow>()
                .AddSingleton<ChecklistWorkflow>()
                .AddSingleton<PaymentWorkflow>()

File: backend/Rashtan.AuditStory.PaypalPayment/ServiceCollectionExtension.cs
using Microsoft.Extensions.DependencyInjection;

namespace Rashtan.AuditStory.PaypalPayment
{
    public static class ServiceCollectionExtension
    {
        public static IServiceCollection RegisterPaypalServices(this IServiceCollection services)
            => services.AddSingleton<IPaypalService, PaypalService>();
    }
}
